### PR TITLE
Prateek&pranay

### PR DESCRIPTION
**Pull request by Rishabh606**

### CODE REVIEW

File: profile/profile-schema-validation.js [hunk 1]
```diff
@@ -12,44 +12,84 @@
...
+  const result = newLawyerProfileSchema.validate(data);
+  if (!result.error) {
+    next();
+  } else {
+    next(createError(400, "bad request"));
+  }
+}
+function updateProfileSchema(req, res, next) {
+  const ParamsToBeUpdated = req.body;
+  let check = 0;
```
Comment (Rishabh606) [inline profile/profile-schema-validation.js]: Do not use variable names like "check". The right var name would be isInvalidKeyPresent and it should be boolean.

File: profile/profile-schema-validation.js [hunk 2]
```diff
@@ -12,44 +12,84 @@
...
+      NameOfField: Joi.string().required(),
+      yearsOfExperience: Joi.number().required(),
+    })
+  ),
+  availableTimeSlots: Joi.array().items(Joi.string()),
+  practiceAreas: Joi.array().items(Joi.string()),
+  awards: Joi.array().items(Joi.string()),
+  clients: Joi.array().items(Joi.string()),
+  address: Joi.array().items(Joi.string()),
+};
```
Comment (Rishabh606) [inline profile/profile-schema-validation.js]: add additional options where we can strip unwanted data from request:
    const options = {
        abortEarly: false, // include all errors
        allowUnknown: true, // ignore unknown props
        stripUnknown: true // remove unknown props
    };
https://jasonwatmore.com/post/2020/07/22/nodejs-express-api-request-schema-validation-with-joi

File: profile/profile-schema-validation.js [hunk 3]
```diff
@@ -12,44 +12,84 @@
...
+  if (!result.error) {
+    next();
+  } else {
+    next(createError(400, "bad request"));
+  }
+}
+function updateProfileSchema(req, res, next) {
+  const ParamsToBeUpdated = req.body;
+  let check = 0;
+  for (let keys in ParamsToBeUpdated) {
```
Comment (Rishabh606) [inline profile/profile-schema-validation.js]: If used earlier advice:
add additional options where we can strip unwanted data from request:
    const options = {
        abortEarly: false, // include all errors
        allowUnknown: true, // ignore unknown props
        stripUnknown: true // remove unknown props
    };
https://jasonwatmore.com/post/2020/07/22/nodejs-express-api-request-schema-validation-with-joi

you would not need to try this check. Unwanted parameters don't affect us.

**Review by Rishabh606 — COMMENTED** [review]
Complete the following changes:
That's to majorly add options in joi which allow to strip useless data as it is harmless but we don't want to save it in db

File: profile/profile-schema-validation.js (hunk 3 above)
Comment (PranayB003) [reply profile/profile-schema-validation.js]: We've made the changes in the latest commit bhaiya, please review and merge

**Pull request merged**